### DISCUSSION TITLE
Add authentication Option

### DISCUSSION THREAD

**Discussion by nWhitehill** [gitpod-io/openvscode-server#156] [Q&A] [answered]
I would love to contribute a new feature where you can deploy this server with the option to have it serve up an Oauth splash screen before allowing a user to go to the workspace. Only authenticated users would be allowed to use the workspace in this use case. Additionally or optionally, adding the option to force a user to provide a password before going to the workspace.

I would love the chance to contribute to this project but want to make sure this feature would be welcomed by the community / maintainers before spending too much time on it.

**Comment by akosyakov** [comment, marked as answer]
Auth is not in the scope. It changes from adopter to adopter. Please see for the scope: https://github.com/gitpod-io/openvscode-server#the-scope-of-this-project

**Comment by andymac4182** [comment]
@akosyakov Would a better option be to look at providing the hooks for people to integrate their own auth. It feels like auth is something that multiple users of this would like to have and would be nice to have a semi standard way of integrating that auth into the project.

**Comment by michaellilltokiwa** [comment]
@andymac4182 what kind of hooks are you thinking about? I don't think it is necessary/reasonable to integrate hooks for auth in this project.

@akosyakov +1 for keeping this as close as possible to the official vscode repo.

**Comment by akosyakov** [comment]
There are already hooks btw, you can provide your own entry point and pass different options to main, here example in Gitpod: https://github.com/gitpod-io/openvscode-server/blob/8f5ea0d6f10ddbbd209763adfb1fe05480bcb7e9/src/vs/gitpod/node/server.ts#L72

Although I don't think you need to bake in auth in server, but build around via a reverse proxy for instance.

**Comment by filiptronicek** [comment]
Simple auth has been added in release 1.62, see https://github.com/gitpod-io/openvscode-server#a-note-about-security-tokens for details.
